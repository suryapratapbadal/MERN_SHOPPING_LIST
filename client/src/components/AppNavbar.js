import React, { Component } from 'react';
import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
    NavItem,
    NavLink,
    Container,
    Button,
} from 'reactstrap';

export default class AppNavbar extends Component {
    constructor(props) {
        super(props);
        this.state = {
            isOpen : false,
        };

    }

    toggle = () => {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }

    render() {
        return (
            <div>
                <Navbar color="dark" dark expand="sm" className="mb-5">
                    <Container>
                        <NavbarBrand href="/">ShoopinList</NavbarBrand>
                        <NavbarToggler onClick={this.toggle} />
                        <Collapse isOpen={this.state.isOpen} navbar>
                            <Nav className="ml-auto" navbar>
                                <NavItem>
                                    <NavLink href="https://github.com/suryapratapbadal/MERN_SHOPPING_LIST">Github</NavLink>
                                </NavItem>
                            </Nav>
                            <Button color='secondary' style={{marginLeft: '2em'}} disabled={!this.props.user}>Log Out</Button>
                        </Collapse>
                    </Container>
                </Navbar>
            </div>
        );
    }
}