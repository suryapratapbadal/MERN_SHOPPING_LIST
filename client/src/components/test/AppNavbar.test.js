import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// Component to be tested
import { AppNavbar } from '../AppNavbar';


const classes = {
  root: {},
  grow: {},
  menuButton: {},
}
const mocklogOutUserfn = jest.fn();
const mockchangeThemefn = jest.fn();

const user = false;

// Test to check , Component Renders
describe('<AppNavbar />', () => {

  const wrapper = shallow(<AppNavbar user={user} logOutUser={mocklogOutUserfn} classes={classes} changeTheme={mockchangeThemefn}/>);

  describe('render()', () => {
    test('renders the component', () => {

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('OnChange()', ()=> {
    test('toggles switch', ()=> {
      wrapper.find('WithStyles(Switch)').props().onChange();
      expect(mockchangeThemefn).toHaveBeenCalled();
      expect(wrapper.state('dark')).toEqual(true);
    });
  });

  describe('onClick()', ()=> {
    test('logs out the user', ()=>{
      wrapper.find('WithStyles(Button)').props().onClick();
      expect(mocklogOutUserfn).toHaveBeenCalled();
    });
  });
});
