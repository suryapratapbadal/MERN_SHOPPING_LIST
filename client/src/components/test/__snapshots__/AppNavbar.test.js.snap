// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppNavbar /> render() renders the component 1`] = `
<div
  className={Object {}}
>
  <WithStyles(AppBar)
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(IconButton)
        aria-label="Menu"
        className={Object {}}
        color="inherit"
      >
        <pure(AddShoppingCartIcon) />
      </WithStyles(IconButton)>
      <WithStyles(Typography)
        className={Object {}}
        color="inherit"
        href="/"
        variant="title"
      >
        ShoopinList
      </WithStyles(Typography)>
      Light
      <WithStyles(Switch)
        checked={false}
        onChange={[Function]}
      />
      Dark
      <WithStyles(Button)
        color="secondary"
        disabled={true}
        onClick={[Function]}
        style={
          Object {
            "marginLeft": "2em",
          }
        }
        variant="contained"
      >
        Log Out
      </WithStyles(Button)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;
